### PR TITLE
Add method to get public Absolute URL.

### PR DESCRIPTION
**Pull request by ChamathNS**
### Proposed changes in this pull request

Rename getAbsoluteURL() to getAbsoluteURLInternal().
Add new method to get the public absolute URL.

improvement for #2851.

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java [hunk 1]
```diff
@@ -84,11 +84,19 @@
...
-     * the fragment to return the absolute URL.
+     * the fragment to return the internal absolute URL.
      *
-     * @return The absolute URL from the Service URL instance.
+     * @return The internal absolute URL from the Service URL instance.
      */
-    String getAbsoluteURL();
+    String getAbsoluteURLInternal();
+
+    /**
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java:93]: A description of how this URL differs from the internal URL would be useful for the API consumer.
Basically when to use this API over the API to get internal URL

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java [hunk 2]
```diff
@@ -84,11 +84,19 @@
...
     /**
      * Concatenate the protocol, host name, port, proxy context path, web context root, url context, query params and
-     * the fragment to return the absolute URL.
+     * the fragment to return the internal absolute URL.
      *
-     * @return The absolute URL from the Service URL instance.
+     * @return The internal absolute URL from the Service URL instance.
      */
-    String getAbsoluteURL();
+    String getAbsoluteURLInternal();
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java:91]: getInternalAbsoluteUrl or getInternalUrl?
@malithie WDYT?

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java (hunk 1 above)
Comment (malithie) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java:94]: Shall we say getAbsoluteInternalURL

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java [hunk 3]
```diff
@@ -83,12 +83,22 @@
...
+    String getInternalAbsoluteURL();
+
+    /**
+     * Returns the proxy server url when the Identity Server is fronted with a proxy.
+     * Concatenate the protocol, host name, port, proxy context path, web context root, url context, query params and
+     * the fragment to return the public absolute URL.
+     *
+     * @return The public absolute URL from the Service URL instance.
+     */
+    String getPublicAbsoluteURL();
```
Comment (malithie) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java:101]: Shall we say getAbsolutePublicURL

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 4]
```diff
@@ -282,7 +282,7 @@ private String buildUrl(String defaultContext, Supplier<String> getValueFromFile
     private String buildTenantQualifiedUrl(String context) {
 
         try {
-            return ServiceURLBuilder.create().addPath(context).build().getAbsoluteURL();
+            return ServiceURLBuilder.create().addPath(context).build().getInternalAbsoluteURL();
```
Comment (malithie) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:285]: I think this should be public URL

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java (hunk 1 above)
Comment (mefarazath) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java:94]: +1

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java (hunk 4 above)
Comment (mefarazath) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:285]: Yes.

**Pull request merged**